gremlin-python/src/main/python/gremlin_python/driver/client.py

```diff
@@ -129,24 +123,28 @@ def _get_connection(self):
             headers=self._headers)
 
     def submit(self, message, bindings=None, request_options=None):
-        return self.submitAsync(message, bindings=bindings, request_options=request_options).result()
+        return self.submit_async(message, bindings=bindings, request_options=request_options).result()
 
-    def submitAsync(self, message, bindings=None, request_options=None):
```

Comment:
I think that renaming this is good but it is a breaking change for users. let's deprecate `submitAsync` for `submit_async` and have both with the old function just calling the new one. When merged to master, i will delete the `submitAsync` one. 

note that we do use `warnings` (at least in one place) for deprecation:

```python
        warnings.warn(
            "As of release 3.3.5, replaced by the gremlin_python.process.anonymous_traversal.traversal() function.",
            DeprecationWarning)
```

I suppose a code comment would suffice as well if you think the warning thing is too heavy

Reply:
I like the idea of warnings, especially when we will be changing code that has been this way for ages and is in production all over the place. I would prefer we keep both for a while rather than remove the old one.

Reply:
That is a good point, I will do the warning / DeprecationWarning thing noted above.